src/App.js

```diff
@@ -1,23 +1,61 @@
-import logo from './logo.svg';
-import './App.css';
+import "./App.css";
+import contactsArr from "./contacts.json";
+import { useState } from "react";
 
 function App() {
+  let fiveContacts = contactsArr.slice(0, 5);
+
+  const [contacts, setContacts] = useState(fiveContacts);
+
+  const restContacts = contactsArr.filter(
+    (contact) => !contacts.includes(contact)
+  );
+
+  const addRandomContact = () => {
+    console.log(restContacts);
+    setContacts(
+      contacts.concat(restContacts[Math.floor(Math.random() * contacts.length)])
+    );
+
+    //setContacts((contactsArr) => {
+    //  const newList = contactsArr.concat(
+    //
+    //  );
+    //  console.log(newList);
+    //  return newList;
+    //});
+  };
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <div>
+        <h1>IronContacts</h1>
+      </div>
+      <div>
+        <button onClick={addRandomContact}>Add Random Contact</button>
+      </div>
+      <table>
+        <thead>
+          <tr>
+            <th>Picture</th>
+            <th>Name</th>
+            <th>Popularity</th>
+            <th>Oscar</th>
+            <th>Emmy</th>
+          </tr>
+        </thead>
+        {fiveContacts.map((contact) => {
+          return (
```

Comment:
You don't map only through your fiveContacts, but through contacts, that will change dynamically, thanks to 
-- > const [contacts, setContacts] = useState(fiveContacts)

"fiveContacts" is only your initial value
e.g. if you add a random contact, then your setContacts function will change again. 

```
 {contacts.map((contact) => .... )}
 
```